### PR TITLE
apply filterStringFunction when customIndexFn

### PR DESCRIPTION
**Pull request by sjhewitt**
fix for #3991

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/3992)

### CODE REVIEW

File: worker/task.go [hunk 1]
```diff
@@ -1388,9 +1389,20 @@ func (qs *queryState) filterStringFunction(arg funcArgs) error {
...
+		filtered = matchStrings(filtered, values, filter)
+	case standardFn:
+		filter.tokens = arg.srcFn.tokens
+		filter.match = defaultMatch
+		filter.tokName = "term"
+		filtered = matchStrings(filtered, values, filter)
+	case customIndexFn:
+		filter.tokens = arg.srcFn.tokens
+		filter.match = defaultMatch
+		filter.tokName = arg.q.SrcFunc.Args[0]
```
[the remark this answers is not in the document]
Comment (sjhewitt) [reply worker/task.go]: I figured this has already been checked by the `parseSrcFn` and must be valid in order to reach this point

**Comment by danielmai** [comment]
> Also noticed that these tests are not being skipped in CI

https://github.com/dgraph-io/dgraph/blob/c6ecca1fe8ca11bcd7bdc8a0a337d2068e503146/test.sh#L58

https://github.com/dgraph-io/dgraph/blob/c6ecca1fe8ca11bcd7bdc8a0a337d2068e503146/test.sh#L198-L201

We've always been setting `-short=true` via test.sh. We can remove so that these tests are run. The only reason only short tests ran was so that the test suite finished quicker. Now that we're running the tests somewhat in parallel, this is less of a concern, so they'll be OK to run now.

I'll open a PR to fix test script for CI.

**Comment by danielmai** [comment]
@sjhewitt TestPlugins now runs in CI with #4002 merged. You can merge the latest master to pick up the test changes.

**Comment by sjhewitt** [comment]
@pawanrawal  looks like the CI tests pass now

**Pull request merged**